Source/WebCore/dom/Element.h

```diff
@@ -505,8 +505,9 @@ class Element : public ContainerNode {
     virtual void didStopBeingFullscreenElement() { }
 
     bool isFinishedParsingChildren() const { return isParsingChildrenFinished(); }
-    void finishParsingChildren() override;
-    void beginParsingChildren() final;
+    // Overriding these functions to make parsing a special case should be avoided if possible.
```

Comment:
I don’t like this comment because it doesn't say why. Performance? Correctness? Inelegant solution to a problem best solved another way?

Reply:
Hm... do you recall what was the original reasoning? The comment was added in https://github.com/WebKit/WebKit/commit/2ee64fe8e40dedcecc1b0948a5cc0a1da51714ac#diff-8d6a42a79982c64a568baee74bc315f946fdccaa399fa1ff272a668db4f85b2eR171.

As far as I know, special casing parser is bad because then elements won't respond to JS DOM mutations the same way but I'm wondering if there was some other reason for it.

Reply:
Updating the comment in https://github.com/WebKit/WebKit/pull/23448